registry_windows.go

```diff
@@ -0,0 +1,122 @@
+package gowsl
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
+)
+
+// guid is a type alias for windows.GUID. Note that syscall.GUID is not
+// equivalent.
+type guid = windows.GUID
+
+// defaultDistro gets the name of the default distribution.
+func defaultDistro() (name string, err error) {
+	defer func() {
+		if err == nil {
+			return
+		}
+		err = fmt.Errorf("failed to obtain default distro: %v", err)
+	}()
+
+	lxssKey, err := registry.OpenKey(lxssRegistry, lxssPath, registry.READ)
+	if err != nil {
+		return "", fmt.Errorf("failed to open lxss registry: %v", err)
+	}
+	defer lxssKey.Close()
+
+	target := "DefaultDistribution"
+	distroDir, _, err := lxssKey.GetStringValue(target)
+	if errors.Is(err, syscall.ERROR_FILE_NOT_FOUND) {
+		return "", errors.New("no default distro")
+	}
+	if err != nil {
+		return "", fmt.Errorf("cannot find %s:%s : %v", lxssPath, target, err)
+	}
+
+	return readRegistryDistributionName(distroDir)
+}
+
+func distroGUIDs() (distros map[string]guid, err error) {
+	lxssKey, err := registry.OpenKey(lxssRegistry, lxssPath, registry.READ)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open lxss registry: %v", err)
+	}
+	defer lxssKey.Close()
+
+	lxssData, err := lxssKey.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("failed to stat lxss registry key: %v", err)
+	}
+
+	subkeys, err := lxssKey.ReadSubKeyNames(int(lxssData.SubKeyCount))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read lxss registry subkeys: %v", err)
+	}
+
+	distros = make(map[string]guid, len(subkeys))
+	for _, key := range subkeys {
+		guid, err := windows.GUIDFromString(key)
```

Comment:
oh nice, there is a built-in for this (even if it’s in `x\`) :)

Reply:
Didn't know that, I guess it works as-is, but if we ever get to make this compilable on Linux we'll probably have to switch in order to hide all `windows` dependencies.

Reply:
Then again this is hidden already in a `_windows` file

Reply:
Yep, actually a syscall to the Win32 API